app/src/hooks/initialize-agent.ts

```diff
@@ -168,11 +197,19 @@ const useInitializeBCAgent = () => {
       return
     }
 
+    const existingAgent = await restartExistingAgent()
+    if (existingAgent) {
+      refreshAttestationMonitor(existingAgent)
+      setAgent(existingAgent)
+      return existingAgent
+    }
+
     const cachedLedgers = await loadCachedLedgers()
     const ledgers = cachedLedgers ?? indyLedgers
 
     const newAgent = await createNewAgent(ledgers)
     if (!newAgent) {
+      logger.error('Failed to create a new agent')
```

Comment:
Should this throw. Wondering if the app can even continue if we get here?

Reply:
Yes, good point. It's kind of impossible to get a falsey value for `newAgent` without createNewAgent throwing itself - the same early return check of `walletSecret` is in both functions - but just incase, I added a throw.

I added a note to refactor that early return check in our tech debt discussion, but I'll keep this PR as small as possible and make that change in a later one.